### DISCUSSION TITLE
How to use Federation in dev mode with Vue 2?

### DISCUSSION THREAD

**Discussion by adaliszk** [originjs/vite-plugin-federation#123] [Q&A] [answered]
I can see that there was some movement around this, like #72, however, when I run a plan and simple Vue 2 host and remote in a dev mode, the /assets/remoteEntry.js exist, but empty. Are there some extra steps needed for it to work?

**Comment by Jiannan-dev** [comment, marked as answer]
The current dev mode only supports the host side, the remote side cannot use dev mode because vite does not allow new chunk in dev mode, so it cannot generate a remoteEntry.js in dev mode

**Reply by adaliszk** [reply]
Is it possible to manually run the remote entry and export it in a custom file? That way I can imagine doing the same /assets/remoteEntry file and just export it through that

**Reply by Jiannan-dev** [reply]
sry,i don't quite understand what you mean, is the generated remoteEntry file copied into dev mode?

**Reply by adaliszk** [reply]
What I was thinking is to add a file manually under the `/assets/remoteEntry` location, import a class or function from the plugin and export out the modules. Essentially forcing vite to build a JS file with the desired content while in dev mode.

It is a workaround, but one I think could work for now.

**Reply by Jiannan-dev** [reply]
I see, but I guess this will still have other problems, such as shared libs and shared components, which are completely different in dev mode than in build mode, where the libs in dev mode will exist in the hidden directory generated by vite instead of in assets

**Reply by adaliszk** [reply]
Agh, then it's way too complicated for a simple solution. I guess will need to do a custom server-side with building the entries separately.
